src/interfacekernels/InterfaceSorptionSievert.C

```diff
@@ -0,0 +1,160 @@
+GenericReal<is_ad>
+InterfaceSorptionSievertTempl<is_ad>::computeQpResidual(Moose::DGResidualType type)
+{
+  // restrict inputs to physically meaningful values to avoid encountering NANs during linear solve
+  const auto small = 1.0e-20;
+  const GenericReal<is_ad> u = std::max(small, _unit_scale * _u[_qp]);
+  const GenericReal<is_ad> u_neighbor =
+      std::max(small, _unit_scale_neighbor * _neighbor_value[_qp]);
+  const GenericReal<is_ad> temperature_limited = std::max(small, _T[_qp]);
+  const auto R = 8.31446261815324; // ideal gas constant (W/(m2-K))
+
+  GenericReal<is_ad> r = 0; // residual
+
+  switch (type)
+  {
+    case Moose::Element:
+    {
+      r = _test[_i][_qp] * _sorption_penalty *
+          (u - _K0 * std::exp(-_Ea / R / temperature_limited) *
+                   std::sqrt(u_neighbor * R * temperature_limited));
+      break;
+    }
+
+    case Moose::Neighbor:
+    {
+      if (_use_flux_penalty)
+        r = _test_neighbor[_i][_qp] * _flux_penalty * _normals[_qp] *
+            (_diffusivity[_qp] * _grad_u[_qp] -
+             (*_diffusivity_neighbor)[_qp] * _grad_neighbor_value[_qp]);
+      else
+        r = _test_neighbor[_i][_qp] * _normals[_qp] * _diffusivity[_qp] * _grad_u[_qp];
+      break;
+    }
+  }
+  return r;
+}
+
+template <>
+Real
+InterfaceSorptionSievertTempl<false>::computeQpJacobian(Moose::DGJacobianType type)
+{
+  // restrict inputs to physically meaningful values to avoid encountering NANs during linear solve
+  const Real small = 1.0e-20;
+  const Real u_neighbor = std::max(small, _unit_scale_neighbor * _neighbor_value[_qp]);
+  const Real temperature_limited = std::max(small, _T[_qp]);
+  const auto R = 8.31446261815324; // ideal gas constant (W/(m2-K))
+
+  Real jac = 0; // jacobian
+
+  switch (type)
+  {
+    case Moose::ElementElement:
+      jac = _test[_i][_qp] * _sorption_penalty * _phi[_j][_qp] * (1);
+      break;
+
+    case Moose::ElementNeighbor:
+      jac = -_test[_i][_qp] * _sorption_penalty * _phi_neighbor[_j][_qp] *
+            (_K0 * std::exp(-_Ea / R / temperature_limited) * std::sqrt(R * temperature_limited) *
+             1 / 2 * std::pow(u_neighbor, -1 / 2));
+      break;
+
+    case Moose::NeighborElement:
+      if (_use_flux_penalty)
+        jac = _test_neighbor[_i][_qp] * _flux_penalty * _normals[_qp] * _diffusivity[_qp] *
+              _grad_phi[_j][_qp];
+      else
+        jac = _test_neighbor[_i][_qp] * _diffusivity[_qp] * _grad_phi[_j][_qp] * _normals[_qp];
+      break;
+
+    case Moose::NeighborNeighbor:
+      if (_use_flux_penalty)
+        jac = -_test_neighbor[_i][_qp] * _flux_penalty * _normals[_qp] *
+              (*_diffusivity_neighbor)[_qp] * _grad_phi_neighbor[_j][_qp];
+      break;
+  }
+  return jac;
+}
+
+template <>
+Real InterfaceSorptionSievertTempl<true>::computeQpJacobian(Moose::DGJacobianType /*type*/)
+{
+  mooseError("In ADInterfaceSorptionSievert: computeQpJacobian incorrectly called from within "
+             "AD calculation");
```

Comment:
```suggestion
  mooseError(name()": in ADInterfaceSorptionSievert, computeQpJacobian incorrectly called from within "
             "AD calculation");
```

Don't think we'd ever hit this, but I'd like to have the user-given name of the object included whenever we use `mooseError` for a better user experience.